packages/core/src/__stampers__/api/web/stamper.ts

```diff
@@ -0,0 +1,238 @@
+    start = bytes.length - 1;
+  }
+  let extraZero = 0;
+  if ((bytes[start]! & 128) == 128) {
+    extraZero = 1;
+  }
+  const res = new Uint8Array(bytes.length - start + extraZero);
+  res.set(bytes.subarray(start), extraZero);
+  return res;
+}
+
+export class IndexedDbStamper implements ApiKeyStamperBase {
+  constructor() {
+    if (typeof window === "undefined") {
+      throw new Error("IndexedDB is only available in the browser");
+    }
+  }
+
+  private async openDb(): Promise<IDBDatabase> {
+    return new Promise((resolve, reject) => {
+      const request = indexedDB.open(DB_NAME, 1);
+      request.onupgradeneeded = (event) => {
+        const db = (event.target as IDBOpenDBRequest).result;
+        if (!db.objectStoreNames.contains(DB_STORE)) {
+          db.createObjectStore(DB_STORE);
+        }
+      };
+      request.onsuccess = () => resolve(request.result);
+      request.onerror = () => reject(request.error);
+    });
+  }
+
+  private async storeKeyPair(
+    publicKeyHex: string,
+    privateKey: CryptoKey,
+  ): Promise<void> {
+    const db = await this.openDb();
+    return new Promise((resolve, reject) => {
+      const tx = db.transaction(DB_STORE, "readwrite");
+      const store = tx.objectStore(DB_STORE);
+      store.put(privateKey, publicKeyHex);
+      tx.oncomplete = () => {
+        db.close();
+        resolve();
+      };
+      tx.onerror = () => reject(tx.error);
+      tx.onabort = () => reject(tx.error);
+    });
+  }
+
+  private async getPrivateKey(publicKeyHex: string): Promise<CryptoKey | null> {
+    const db = await this.openDb();
+    return new Promise((resolve, reject) => {
+      const tx = db.transaction(DB_STORE, "readonly");
+      const store = tx.objectStore(DB_STORE);
+      const request = store.get(publicKeyHex);
+      request.onsuccess = () => {
+        db.close();
+        resolve(request.result || null);
+      };
+      request.onerror = () => {
+        db.close();
+        reject(request.error);
+      };
+    });
+  }
+
+  async listKeyPairs(): Promise<string[]> {
+    const db = await this.openDb();
+    return new Promise((resolve, reject) => {
+      const tx = db.transaction(DB_STORE, "readonly");
+      const store = tx.objectStore(DB_STORE);
+      const request = store.getAllKeys();
+      request.onsuccess = () => {
+        db.close();
+        resolve(request.result as string[]);
+      };
+      request.onerror = () => {
+        db.close();
+        reject(request.error);
+      };
+    });
+  }
+
+  async createKeyPair(externalKeyPair?: CryptoKeyPair): Promise<string> {
+    let privateKey: CryptoKey;
+    let publicKey: CryptoKey;
+    if (externalKeyPair) {
+      const extractable = (externalKeyPair.privateKey as any).extractable;
+      if (extractable !== false) {
+        throw new Error("Provided privateKey must be non-extractable.");
+      }
+      privateKey = externalKeyPair.privateKey;
+      publicKey = externalKeyPair.publicKey;
```

Comment:
We should validate that  the private and public key provided here represent a valid P256 key pair